mongo/datadog_checks/mongo/mongo.py

```diff
@@ -176,39 +168,39 @@ def _build_metric_list_to_collect(self):
         return metrics_to_collect
 
     def _refresh_replica_role(self):
-        if self._api_client and isinstance(self._api_client.deployment_type, ReplicaSetDeployment):
+        if self._api_client and (
+            self._api_client.deployment_type is None
+            or isinstance(self._api_client.deployment_type, ReplicaSetDeployment)
+        ):
             self.log.debug("Refreshing deployment type")
             self._api_client.deployment_type = self._api_client.get_deployment_type()
 
     def check(self, _):
-        try:
+        if self._connect():
             self._check()
-        except (pymongo.errors.ConnectionFailure, Exception):
-            self._api_client = None
-            raise
 
-    def _check(self):
-        try:
-            api = self.api_client
-        except Exception as e:
-            self.service_check(SERVICE_CHECK_NAME, AgentCheck.CRITICAL, tags=self._config.service_check_tags)
-            self.log.exception("Error when creating the api client: %s.", str(e))
-            raise
+    def _connect(self) -> bool:
+        if self._api_client is None:
+            try:
+                self._api_client = MongoApi(self._config, self.log)
+                self.log.debug("Connecting to '%s'", self._config.hosts)
+                self._api_client.connect()
+                self.log.debug("Connected!")
+                self._mongo_version = self.api_client.server_info().get('version', '0.0')
+                self.set_metadata('version', self._mongo_version)
+            except Exception as e:
+                self._api_client = None
+                self.log.debug('Exception: %s', e)
```

Comment:
Should we log it as an error? If the customer provide invalid credentials (for instance) they will have no idea what's going on under the hood, they would only have the service check set to critical unless they switch to debug mode.